bin/gwdetchar-scattering

```diff
@@ -347,6 +347,8 @@ for i, channel in enumerate(sorted(alldata[0].keys())):
     else:
         effdt = efficiencypc/deadtimepc
     logger.info("Efficiency/Deadtime: %.2f" % effdt)
+        
+    a.append(scatter_segments[channel].active)
```

Comment:
@siddharth101, I think it's simpler to only append here if the segments are non-empty:

```suggestion
    if abs(scatter_segments[channel].active):
        actives.append(scatter_segments[channel].active)
```

This means you can do away with line 503 completely.

Reply:
Yeah, I was trying to find a way to do that. I will change this.